beamline.rst

```diff
@@ -1,212 +1,379 @@
-        msg: str
-        state: ActuatorState
-        limit: [float, float]
+        # NB value_list, if not empty, gives the allowed values.
+        #    For TYP float a set_value will default to the closest value in the
+        #    values_list.
+        # In all other cases setting a disallowed value should throw ValueError.
+        """
+
+        name:str            # A unique name that identifies this actuator
+        value:Optional[TYP] # The current position - could be None is some states.
+        msg:Optional[str]   # A message string, explaining state or value
+        state:ActuatorState # The state of the actuator
+        upper_limit:Optional[TYP]   # Upper limit
+        lower_limit:Optional[TYP]   # Lower limit
+        value_list:Tuple[TYP]        # Tuple of allowed values
 
 
     def get_actuators() -> Dict[str, ActuatorData]:
-    """
-    :returns: A dictionary with all available actuators where the key
-              is the actuator name and the value the ActuatorData tuple
-    :rtype dict:            
-    """
+        """
+        :returns: A dictionary with all available actuators where the key
+                  is the actuator name and the value the ActuatorData tuple
+        :rtype dict:
+        """
         pass
 
 
     def get_actuator(name) -> ActuatorData:
-    """
-    :returns: The ActuatorData object identified by the given name
-    :rtype: ActuatorData
-    """
-       pass
+        """
+        :returns: The ActuatorData object identified by the given name
+        :rtype: ActuatorData
+        """
+        pass
 
 
     def set_actuator_value(name, value:Any) -> bool:
-    """
-    Tries to move the actuator identified by name to value
-
-    (Errors and progress of movement is passed asynchronously
-     via the available signaling mechanism)
-    
-    :returns: True if motion was started False otherwise
-    """
+        """
+        Tries to set the actuator identified by name to value.
+        Setting a disallowed value will raise ValueError, with one exception:
+        if the actuator takes a float value and has a non-empty values_list,
+        the value will be set to the nearest value in the list.
+
+        (Errors and progress of movement is passed asynchronously
+         via the available signalling mechanism)
+
+        :returns: True if motion was started False otherwise
+        """
         pass
 
+  **State/Value enumerations**
+
+      class ActuatorState(Enum):
+
+        # This enumeration should be limited to what the UI needs to know,
+        # not what the motors might want to tell. These values may need fixing.
+        # Why do we need MOVESTARTED, for instance?
+
+        NOTINITIALIZED = 0  # Actuator has not yet been set up. value is None
+        UNUSABLE = 1        # Actuator is not functional. value is None
+        READY = 2           # Actuator is functional and ready to accept new moves.
+        MOVESTARTED = 3     # Move has started. Why do we need this?
```

Comment:
I don't think that its actually needed, I can imagine that it have been or perhaps are used in cases where it takes a long time between an "inital" move and "actual" move. We are not using this in our UIs as far as I know.